### PR TITLE
[SPARK-25736][SQL][TEST] add tests to verify the behavior of multi-column count

### PR DESCRIPTION
**Pull request by cloud-fan**
## What changes were proposed in this pull request?

AFAIK multi-column count is not widely supported by the mainstream databases(postgres doesn't support), and the SQL standard doesn't define it clearly, as near as I can tell.

Since Spark supports it, we should clearly document the current behavior and add tests to verify it.

## How was this patch tested?

N/A

### CODE REVIEW

**Comment by cloud-fan** [comment]
cc @gatorsmile @mgaido91 @viirya

**Comment by mgaido91** [comment]
this is indeed the behavior I'd expect. Good to add tests to enforce the behavior. Did you check other RDBMs apart from Postgres?

**Comment by cloud-fan** [comment]
I'm going to try Hive and Presto, but my local environment has some problems and I need to fix it first. Will work on it tomorrow.

**Comment by cloud-fan** [comment]
BTW MySQL doesn't support `count(a, b)` but supports `count(distinct a, b)`, the result is same as Spark.

**Comment by viirya** [comment]
Yea, it is definitely good to add document and test for current behavior.

File: sql/core/src/test/resources/sql-tests/inputs/count.sql [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+-- distinct count with single expression
+SELECT
+  count(DISTINCT a),
+  count(DISTINCT b),
+  count(DISTINCT (a + b)),
+  count(DISTINCT (a, b))
+FROM testData;
+
+-- count with multiple expressions
+SELECT count(a, b), count(b, a), count(testData.*) FROM testData;
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/count.sql]: Please also include `count(*)`

File: sql/core/src/test/resources/sql-tests/inputs/count.sql [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+-- distinct count with multiple expressions
+SELECT count(DISTINCT a, b), count(DISTINCT b, a), count(DISTINCT testData.*) FROM testData;
```
Comment (gatorsmile) [inline sql/core/src/test/resources/sql-tests/inputs/count.sql]: Also include `count(DISTINCT *)`

**Comment by gatorsmile** [comment]
Let us add one more case in the test suite. 
```SQL
SELECT count(1), count(NULL) FROM testData;
```

**Comment by gatorsmile** [comment]
LGTM except the above comments.

**Review by HyukjinKwon — APPROVED** [review]

**Comment by viirya** [comment]
LGTM

**Comment by HyukjinKwon** [comment]
Merged to master and branch-2.4.

**Comment by HyukjinKwon** [comment]
(From https://github.com/apache/spark/pull/22773#issuecomment-432917994) @gatorsmile and @cloud-fan, let's say this will break `DESCRIBE FUNCTION EXTENDED`. Should we update migration guide as well?

**Pull request closed without merging**